pdns/packethandler.cc

```diff
@@ -880,11 +880,21 @@ void PacketHandler::addNSEC3(DNSPacket& p, std::unique_ptr<DNSPacket>& r, const
 
   if (d_sd.db == nullptr) {
     if(!B.getSOAUncached(d_sd.qname, d_sd)) {
-      DLOG(g_log<<"Could not get SOA for domain");
+      DLOG(g_log<<"Could not get SOA for domain"<<endl);
       return;
     }
   }
 
+  if (!d_sd.db->doesDNSSEC()) {
```

Comment:
untested, but I think:

```suggestion
  if (!d_sd.db->doesDNSSEC() && !narrow) {
```

Reply:
Makes sense.